Fix runtime error in aiWorker when pricePerUnit is unset

eliteprox opened the pull request:
**What does this pull request do? Explain your changes. (required)**
<!-- A clear and concise description of what this pull request does. -->
Fixes a fatal runtime error when `-pricePerUnit` is not set, which is allowed with `-aiWorker` flag in #3047


**Specific updates (required)**
<!--- List out all significant updates your code introduces -->
- Checks if basePrice is nil before comparing if it is equal to zero. 
- I also added a check if transcodePrice is nil before using it, just in case that condition occurs. Though it shouldn't happen because of startup checks that require `-pricePerUnit` when `-aiWorker` is not set

**How did you test each of these updates (required)**
<!-- A detailed description of how you tested your code changes. Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc. -->
- Verified that ai jobs process at the correct price from aiModels.json
- Verified `-pricePerBroadcaster` prices are set correctly for transcoding work
- Verified `-pricePerUnit` price is used for transcode jobs when `-pricePerBroadcaster` is unset for the broadcasting address
- Verified that `-autoAdjustPrice` still works correctly in all cases

**Does this pull request close any open issues?**
<!-- Fixes # -->
#3057

**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] Read the [contribution guide](./CONTRIBUTING.md)
- [ ] `make` runs successfully
- [ ] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

eliteprox committed:
Fix nil baseprice when pricePerUnit is unused in aiWorker

rickstaa commented on core/orchestrator.go:
```diff
@@ -359,7 +359,7 @@ func (orch *orchestrator) priceInfo(sender ethcommon.Address, manifestID Manifes
 		// If no priced capabilities were signaled by the broadcaster assume that they are requesting
 		// transcoding and set the base price to the transcode price
-		if basePrice.Cmp(big.NewRat(0, 1)) == 0 {
+		if transcodePrice != nil && (basePrice == nil || basePrice.Cmp(big.NewRat(0, 1)) == 0) {
```
@eliteprox, as you mentioned earlier, `basePrice` will never be `nil` because of the following line in the `orchestrator.go` file:

[Orchestrator code](https://github.com/livepeer/go-livepeer/blob/a1b9f33d7261b798d0d33685992a4f93ec4d3473/core/orchestrator.go#L340).

Given this, I believe the condition `basePrice == nil` should be removed. This will ensure that the `basePrice` is correctly set to the `transcodePrice` when `basePrice` is `0`.

Here is the updated code snippet:

```suggestion
		if transcodePrice != nil && basePrice.Cmp(big.NewRat(0, 1)) == 0 {
```

This change ensures that the `basePrice` will be set to `transcodePrice` when it is zero, reflecting the intended behavior more accurately.

rickstaa committed:
fix: fix priceInfo 'nil' error on discovery

This commit ensures that when the `transcodePrice` is not set by the AI
orchestrator no `nil` error is thrown when a Gateway requests the
orchestrators OrchInfo.

rickstaa committed:
fix(ai): fix incorrect transcodePrice condition

This commit fixes the check that is performed to check if transcodePrice
is set.

eliteprox committed:
Merge pull request #1 from livepeer/fix-nil-baseprice2

Tested transcoding with `-pricePerUnit` and `-pricePerBroadcaster`, verified the correct price comes through. 

Tested AI jobs with and without pricePerUnit, in every case the price from the model is used and no nil errors occur. 

Tested without `-autoAdjustPrice=false` so that it would reach the line it was previously breaking on. Everything looks good.
